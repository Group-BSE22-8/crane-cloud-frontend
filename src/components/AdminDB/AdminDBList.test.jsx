/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import AdminDBList, { mapStateToProps } from "./";

const AdminDBListProps = {
  database: { name: "sanlam" },
  params: { projectID: "1" },
  message: "This successful",
  adminGetDatabases: jest.fn(),
  showCreateComponent: jest.fn(),
  callbackCreateComponent: jest.fn(),
};

describe("Test AdminDBList component", () => {
  let dbListWrapper;
  beforeEach(() => {
    const AdminDBListwrapper = AdminDBList.WrappedComponent;
    dbListWrapper = shallow(<AdminDBListwrapper {...AdminDBListProps} />);
  });

  it("should check `componentDidMount()`", () => {
    dbListWrapper.instance().componentDidMount();
    expect(AdminDBListProps.adminGetDatabases).toHaveBeenCalledTimes(2); 
    
    expect(AdminDBListProps.adminGetDatabases).toBeDefined();
    expect(AdminDBListProps.params).toBeDefined();
    expect(dbListWrapper.instance().props.adminGetDatabases).toBeCalled();
  });
  it("matching component snapshot for AdminDBList", () => {
    expect(dbListWrapper).toMatchSnapshot();
  });
});

describe("Test AdminDBList's map state to props and dispatch", () => {
  it("test result to match the mapstostate function", () => {
    expect(
      mapStateToProps({
        adminDatabasesReducer: { isFetchingDatabases: false, databases: [], databasesFetched: false },
        adminCreateDBReducer: { isCreated: false}
      })
    ).toEqual({
      isFetchingDatabases: false,
      databases: [],
      databasesFetched: false,
      isCreated: false,
    });
  });
});
