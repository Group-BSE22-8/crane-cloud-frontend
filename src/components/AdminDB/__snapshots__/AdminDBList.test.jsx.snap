// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AdminDBList component matching component snapshot for AdminDBList 1`] = `ShallowWrapper {}`;
